pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -617,6 +617,28 @@ public boolean equals(Object o) {
         .isEqualIgnoreOrder(_dateTimeFieldSpecs, that._dateTimeFieldSpecs);
   }
 
+  public boolean isBackwardCompatibleWith(Schema oldSchema) {
+    if (oldSchema == null) {
```

Comment:
In the current code path it's not null. Still feeling as a generic util in `Schema` class, it's good to always check this.

Reply:
IMO, this should throw NPE (similar to `compareTo()`). User should check whether the schema is null before calling this method. If user code has some bug and accidentally call this method with null, we should terminate the operation instead of letting them put a new schema.

Reply:
Makes sense. Updated.